src/System.Linq/src/System/Linq/Buffer.cs

```diff
@@ -2,15 +2,14 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-using System;
 using System.Collections.Generic;
 
 namespace System.Linq
 {
     internal struct Buffer<TElement>
     {
-        internal TElement[] items;
-        internal int count;
+        internal readonly TElement[] _items;
+        internal readonly int _count;
```

Comment:
_items is a reference type, _count is a primitive and we do not call its methods anyways. 
I did check for the cases where readonly structs would introduce unnecessary copying. I do not thing we had any such case in this change.